syntax = "proto3";
package pgml;

import "google/protobuf/descriptor.proto";

extend google.protobuf.FileOptions {
	optional bool enable = 1091;
}

message ArgRef {
    string name = 1;
    string path = 2;
}

message ArgRefs {
    repeated ArgRef item = 1;
}

extend google.protobuf.MessageOptions {
	// Python import path for a class to instantiate using the values
	// of this message.
	string builder = 1091;
    // Map of argument names to paths. The path is the dotted path to the argument
    // from the root of the CLI args class.
    ArgRefs arg_refs = 1092;
}

// Defaults values applied at the field level
extend google.protobuf.FieldOptions {
	// Value specify the default value to set on this field. By default,
	// none is set on a field.
	optional FieldDefaults default = 1171;
	// If true, and builder is set, the field will be treated as a varargs field.
	bool as_varargs = 1172;
}

// FieldDefaults encapsulates the default values for each type of field. Depending on the
// field, the correct set should be used to ensure proper defaults generation.
message FieldDefaults {
	oneof type {
		// Scalar Field Types
		float float = 1;
		double double = 2;
		int32 int32 = 3;
		int64 int64 = 4;
		uint32 uint32 = 5;
		uint64 uint64 = 6;
		sint32 sint32 = 7;
		sint64 sint64 = 8;
		fixed32 fixed32 = 9;
		fixed64 fixed64 = 10;
		sfixed32 sfixed32 = 11;
		sfixed64 sfixed64 = 12;
		bool bool = 13;
		string string = 14;
		bytes bytes = 15;
		uint32 enum = 16;
	}
	reserved 18 to 20;
}
